load-tsfile-tool/src/main/java/org/apache/iotdb/TsFileLoaderTool.java

```diff
@@ -0,0 +1,407 @@
+      for (File file : unloadTsFiles) {
+        System.out.println(file.getPath());
+      }
+    }
+  }
+
+  /**
+   * Read a TsFile and write into IoTDB session. This method can load TsFile with IoTDB version.
+   * Support TsFile generated from IoTDB version 0.12 - 0.14(including Aligned Timeseries).
+   *
+   * @param filename the file path to be loaded
+   * @param session IoTDB session
+   */
+  public static void writeTsFile(String filename, Session session)
+      throws IOException, IllegalPathException, IoTDBConnectionException,
+          StatementExecutionException, NoMeasurementException {
+    // parse modifications from .mods
+    List<Modification> modifications = null;
+    if (FSFactoryProducer.getFSFactory()
+        .getFile(filename + ModificationFile.FILE_SUFFIX)
+        .exists()) {
+      modifications =
+          (List<Modification>)
+              new ModificationFile(filename + ModificationFile.FILE_SUFFIX).getModifications();
+    }
+
+    // read all device and their measurements
+    parseDeviceFromTsFile(filename);
+
+    try (TsFileSequenceReader reader = new TsFileSequenceReader(filename)) {
+      for (Map.Entry<String, Set<MeasurementSchema>> entry : device2Measurements.entrySet()) {
+        // collect measurements for device
+        boolean isAligned = false;
+        String curDevice = entry.getKey();
+        List<MeasurementSchema> measurementSchemas = new ArrayList<>();
+        ArrayList<Path> paths = new ArrayList<>();
+        for (MeasurementSchema measurementSchema : entry.getValue()) {
+          if (!measurementSchema.getType().equals(TSDataType.VECTOR)) {
+            measurementSchemas.add(measurementSchema);
+          } else {
+            isAligned = true;
+          }
+        }
+        for (MeasurementSchema measurementSchema : measurementSchemas) {
+          paths.add(new Path(curDevice, measurementSchema.getMeasurementId()));
+        }
+
+        // construct query to this tsfile
+        List<AbstractFileSeriesReader> readersOfSelectedSeries = new ArrayList<>();
+        List<TSDataType> dataTypes = new ArrayList<>();
+        IMetadataQuerier metadataQuerier = new MetadataQuerierByFileImpl(reader);
+        IChunkLoader chunkLoader = new CachedChunkLoaderImpl(reader);
+        for (Path path : paths) {
+          List<IChunkMetadata> chunkMetadataList = metadataQuerier.getChunkMetaDataList(path);
+          modifyChunkMetadata(isAligned, path, chunkMetadataList, modifications);
+          AbstractFileSeriesReader seriesReader;
+          if (chunkMetadataList.isEmpty()) {
+            seriesReader = new EmptyFileSeriesReader();
+            dataTypes.add(metadataQuerier.getDataType(path));
+          } else {
+            seriesReader = new FileSeriesReader(chunkLoader, chunkMetadataList, null);
+            dataTypes.add(chunkMetadataList.get(0).getDataType());
+          }
+          readersOfSelectedSeries.add(seriesReader);
+        }
```

Comment:
It seems that you read each sub sensor of an aligned device seperately, but it's better and more efficient to read them together.

Reply:
Actually,
```
QueryDataSet dataSet =
            new DataSetWithoutTimeGenerator(paths, dataTypes, readersOfSelectedSeries);
```
is the real query statement, so aligned timeseries is queried by aligned API.